criu/pstree.c

```diff
@@ -384,6 +384,11 @@ static int prepare_pstree_for_shell_job(pid_t pid)
 				pi->sid = current_sid;
 		}
 
+		for_each_pstree_item(pi) {
+			if (pi->pgid == old_sid)
+				pi->pgid = current_sid;
+		}
+
```

Comment:
Would it be possible to combine both loops into a single one?

```C
for_each_pstree_item(pi) {
  if (pi->sid == old_sid)
    pi->sid = current_sid;

  if (pi->pgid == old_sid)
    pi->pgid = current_sid;
}
```

Reply:
Sure, done.